src/main/java/com/google/devtools/build/lib/analysis/platform/PlatformInfo.java

```diff
@@ -179,6 +181,19 @@ public static Builder builder() {
     return new Builder();
   }
 
+  /**
+   * Add this platform to the given fingerprint.
+   */
+  public void addTo(Fingerprint f) {
+    f.addString(label.toString());
+    f.addStrings(constraints.values().stream()
+      .map(c -> c.label().toString())
+      .collect(Collectors.toList()));
+    if (remoteExecutionProperties != null) {
```

Comment:
This way, you can get the same strings added to the fingerprint if you remove remoteExecutionProperties but add a string to constraints.values(). Instead, use:

f.addString(label.toString())
f.addNullableString(remoteExecutionProperties);
f.addInt(constraints.size());
<add all the values>

Reply:
Also, why not add the keys of the map?

Reply:
The keys are actually based on the values, and only exist to make lookup efficient. The actual rule is only determines by the list of constraint values.

Updated the method.